### PR TITLE
Added advanced stun baton that HoS now starts with.

### PR DESCRIPTION
**Pull request by twirling-jesters**
dreammaker_2020-07-13_00-56-26

I feel like providing the HoS a go-to melee weapon that acts as just a slightly stronger, more efficient version of the stun baton would be worthwhile. Giving the HoS a "better baton" was something that telebatons tried to solve, but in my experience heads of security usually just opt for the standard stun baton due to its duration. This is just a slight bump up in duration (now about 4 seconds or so) which uses about half the power of a normal baton. Granted, this is my first pull request and the wiki told me to be conservative with my changes, so I didn't want to do too much to change from the standard baton model.

:cl:
rscadd: Added a slightly stronger version of the basic stunbaton that starts in the HoS's backpack. Can be disassembled by research for decent combat research.

### CODE REVIEW

**Comment by Dacendeth** [comment]
30 second stun baton? uhhh pass.

**Comment by Kharkov72** [comment]
post the icon eggman

File: code/game/objects/items/weapons/stunbaton.dm [hunk 1]
```diff
@@ -259,6 +259,20 @@
...
 
+//Advanced stun baton. Currently in testing.
+/obj/item/weapon/melee/baton/advbaton
+	name = "Advanced Stun Baton"
+	desc = "A more powerful version of the standard-issue stun baton."
+	icon_state = "adv_baton"
+	hitcost = 50
+	stunforce = 15
+
+/obj/item/weapon/melee/baton/advbaton/New() //Pre-installing cell.
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/stunbaton.dm]: this is all inherited, remove it

**Review by Kurfursten — CHANGES_REQUESTED** [review]
inheritance

**Comment by boy2mantwicethefam** [comment]
It can't be worse than the battering ram
or the SPS
or the flashbangs
or the shotgun
or the vector
or the 2 minute cuffs
or beepskies

**Comment by boy2mantwicethefam** [comment]
oh and batons can still be equipped with batteries that make them last literally hundreds of stuns before needing to recharge, while prods drain high capacity cells in 6 hits and only stun for 10 seconds. It really can't be worse than what there already is

**Comment by boy2mantwicethefam** [comment]
I didn't mention the lawgiver because fun should be allowed

**Comment by DrSnips** [comment]
Could you explain a few things and maybe update the OP with some of them?
* Why is this item needed?
* Why does it start in the backpack rather than REPLACE his stun baton in his gear box in his locker?
* What's going on with the sprite. I thought the active state had an eletric effect and was lit/lighter at the tip. If it's just an issue with icon diff bot here just say so.

**Comment by twirling-jesters** [comment]
> * Why is this item needed?

I feel like providing the HoS a go-to melee weapon that acts as just a slightly stronger, more efficient version of the stun baton would be worthwhile. Giving the HoS a "better baton" was something that telebatons tried to solve, but in my experience heads of security usually just opt for the standard stun baton due to its duration. This is just a slight bump up in duration (now about 4 seconds or so) which uses about half the power of a normal baton. Granted, this is my first pull request and the wiki told me to be conservative with my changes, so I didn't want to do too much to change from the standard baton model.

> * Why does it start in the backpack rather than REPLACE his stun baton in his gear box in his locker?

It would feel off-brand to put a non-standard, advanced item into a standard sec gear box. Plus, this prevents compounding urges to break into his office by not stuffing his locker with more high-value equipment. Replacing the telebaton in the locker was an option, but I feel like that still has a niche and should be left as a choice to the HoS.

> * What's going on with the sprite. I thought the active state had an eletric effect and was lit/lighter at the tip. If it's just an issue with icon diff bot here just say so.

This one might be with the icon diff bot, the active sprite for the advanced baton has the halo of light and sparks on my end.

Thank you for your questions/feedback. I've updated accordingly.

**Comment by Gazathor** [comment]
why does the HoS need this, he has funny shoot gun.

**Comment by GreatBigFailure** [comment]
So how long exactly does this version last? I like the concept but just "longer stun" seems a bit unfun if it lasts a super long amount.. 
Maybe make it have a 100% chance to stun when thrown or something similar.

**Comment by DrSnips** [comment]
> Throwing your stun baton ever

Why would you ever do this?

**Comment by Theophrastus1** [comment]
Wow, what a useless addition to the game. Surely you couldn't be adding this for personal reasons.

**Comment by Theophrastus1** [comment]
> 
> 
> > Throwing your stun baton ever
> 
> Why would you ever do this?

For the sick takedowns to go in your ownage montage you play in the breakroom

**Pull request closed without merging**